.github/shared/src/spec-model.js

```diff
@@ -217,3 +231,141 @@ function readme(file) {
+}
+
+/**
+ * Get operations from all previous versions that also exist in current swagger
+ * @param {string} targetSwaggerPath - Path to the current swagger file
+ * @param {Swagger[]} availableSwaggers - Array of swagger objects to search within
+ * @returns {Promise<Map<string, Operation[]>>} - Operations that exist in previous or current versions(but not same swagger)
+ */
+export async function getExistedVersionOperations(targetSwaggerPath, availableSwaggers) {
+  /** @type {Map<string, Operation[]>} */
+  let result = new Map();
+
+  if (!availableSwaggers || !Array.isArray(availableSwaggers)) {
+    return result;
+  }
+
+  const targetSwagger = availableSwaggers.find((s) => s.path === targetSwaggerPath);
+  if (!targetSwagger) {
+    return result;
+  }
+
+  const fileName = await targetSwagger.getFileName();
+  const currentVersion = await targetSwagger.getVersion();
+
+  // Get operations from current swagger file
+  const currentOperationsMap = await targetSwagger.getOperations();
+  const currentOperations = Array.from(currentOperationsMap.values());
+
+  // Load version info for all swaggers to enable filtering
+  const swaggersWithVersions = await Promise.all(
+    availableSwaggers.map(async (swagger) => ({
+      swagger,
+      version: await swagger.getVersion(),
+      fileName: await swagger.getFileName(),
+    })),
+  );
+
+  // Get all current and previous versions of the swaggers with different fileNames
+  const previousVersionSwaggers = swaggersWithVersions
+    .filter((item) => item.fileName !== fileName && item.version <= currentVersion)
+    .sort((a, b) => a.version.localeCompare(b.version))
+    .map((item) => item.swagger);
+
+  // Collect operations from all previous versions that also exist in current version
+  for (const previousSwagger of previousVersionSwaggers) {
+    try {
+      const previousOperationsMap = await previousSwagger.getOperations();
+      const previousOperations = Array.from(previousOperationsMap.values());
+
+      // Find operations that exist in both this previous version AND current version
+      const matchingOperations = previousOperations.filter((operation) => {
+        return currentOperations.some(
+          (currentOp) =>
+            currentOp.path === operation.path &&
+            currentOp.id === operation.id &&
+            currentOp.httpMethod === operation.httpMethod,
+        );
+      });
+      result.set(previousSwagger.path, matchingOperations);
+    } catch {
+      // Skip versions that can't be processed
+    }
+  }
+
+  return result;
+}
+
+/**
+ * Get preceding swagger versions for a given swagger file
+ * @param {string} targetSwaggerPath - Path to the swagger file to find predecessors for
+ * @param {Swagger[]} availableSwaggers - Array of swagger objects to search within
+ * @returns {Promise<{preview?: string, stable?: string}>} - Paths to preceding versions
+ */
+export async function getPrecedingSwaggers(targetSwaggerPath, availableSwaggers) {
```

Comment:
Why does this take `availableSwaggers` as a parameter, rather than using the `getSwaggers()` property of this object?  The point of spec-model is users should just need a pointer to any object in a spec-model graph.  Users shouldn't need to keep their own list of swaggers.  The list should be used from a specModel object, like SpecModel, Readme, Tag, etc.

Reply:
Also, I still think this method (and `getExistedVersionOperations()`) make more sense on the Swagger object, in which case they should need **no** parameters.  The `targetSwaggerPath()` is Swagger.path, and availableSwaggers can be found by walking up from Swagger to Readme or SpecModel (as high as needed).

Reply:
If you disagree, what might help convince me, is adding some real-world examples to this PR of how you want to use the new APIs.